pkg/permissions/rule.go

```diff
@@ -0,0 +1,83 @@
+package permissions
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+const ruleSep = " "
+
+const valueSep = ","
+const partSep = ":"
+
+// Rule represent a single permissions rule, ie a Verb and a type
+type Rule struct {
+	// Type is the JSON-API type or couchdb Doctype
+	Type string `json:"type"`
+
+	// Title is a human readable (i18n key) header for this rule
+	Title string `json:"-"`
+
+	// Description is a human readable (i18n key) purpose of this rule
+	Description string `json:"description,omitempty"`
+
+	// Verbs is a subset of http methods.
+	Verbs VerbSet `json:"verbs,omitempty"`
+
+	// Selector is the field which must be one of Values.
+	Selector string   `json:"selector,omitempty"`
+	Values   []string `json:"values,omitempty"`
+}
+
+// MarshalScopeString transform a Rule into a string of the shape
+// io.cozy.files:GET:io.cozy.files.music-dir
+func (r Rule) MarshalScopeString() (string, error) {
+	out := r.Type
+	if len(r.Verbs) != 0 {
+		out += partSep + r.Verbs.String()
+	}
+	if len(r.Selector) != 0 {
+		out += partSep + r.Selector
+	}
+	if len(r.Values) != 0 {
+		out += partSep + strings.Join(r.Values, valueSep)
+	}
+	return out, nil
```

Comment:
If a rule comes from JSON, it can have a selector and values but not verbs. See https://github.com/aenario/cozy-stack/blob/b7bbb6f1fdbc00c1090e245f7dc764d7f9aade28/pkg/permissions/permissions_test.go#L68-L73 for an example. In that case, we should insert the `ALL` for verbs